internal/controller/util/cephfs_cg.go

```diff
@@ -327,7 +327,7 @@ func GetVolumeSnapshotsOwnedByVolumeGroupSnapshot(
 
 	for _, snapshot := range volumeSnapshotList.Items {
 		for _, owner := range snapshot.ObjectMeta.OwnerReferences {
-			if owner.Kind == "VolumeGroupSnapshot" && owner.Name == vgs.Name {
+			if owner.Kind == "VolumeGroupSnapshot" && owner.Name == vgs.Name && owner.UID == vgs.UID {
```

Comment:
The entire `if` statement could be simplified to `if owner.UID == vgs.UID`, but I prefer the added clarity of the more detailed version.